### PR TITLE
Send the access token along with the id token when creating an account from Google

### PR DESCRIPTION
**Pull request by Tug**
Requires server patch D6682.

This PR updates social login and social sign up (as well as social linking) to include the access_token given by google in the request payload.
This is necessary to fetch the user's avatar on the server on signup and might serve other purposes in the future.

### Social signup from login page
- Go to https://plus.google.com/u/0/me and upload an avatar if you don't already have one
- Go to http://calypso.localhost:3000/log-in and click on "Continue with Google"
- Authenticate with the same google account
- Follow to the server patch instruction to execute the avatar update job
- Go to http://calypso.localhost:3000/me and check that `First name`, `Last Name` and `Public Display Name` are filled. Check that your google avatar has been uploaded

#### Test that all other flows still work
- Follow to the server patch instruction to unlink your google account and/or reset your password
- Social signup at http://calypso.localhost:3000/start/social
- Test social linking
- Test social login

### Reviews
- [x] Product
- [x] Code

### CODE REVIEW

**Comment by scruffian** [comment]
I tested this and it works as expected.

File: client/blocks/login/social.jsx [hunk 1]
```diff
@@ -40,11 +40,17 @@ class SocialLoginForm extends Component {
...
 
-		if ( ! response.Zi || ! response.Zi.id_token ) {
+		if ( ! response.Zi || ! response.Zi.access_token || ! response.Zi.id_token ) {
 			return;
 		}
 
-		this.props.loginSocialUser( 'google', response.Zi.id_token, redirectTo )
+		const socialInfo = {
+			service: 'google',
+			access_token: response.Zi.access_token,
```
Comment (yurynix) [inline client/blocks/login/social.jsx]: nitpick: shouldn't we use camelCase?

File: client/lib/wpcom-undocumented/lib/undocumented.js [hunk 2]
```diff
@@ -1436,26 +1436,20 @@ Undocumented.prototype.usersNew = function( query, fn ) {
...
+	query.locale = i18n.getLocaleSlug();
 
 	// This API call is restricted to these OAuth keys
-	restrictByOauthKeys( body );
+	restrictByOauthKeys( query );
 
 	const args = {
 		path: '/users/social/new',
-		body
+		body: query
```
Comment (yurynix) [inline client/lib/wpcom-undocumented/lib/undocumented.js]: I'm confused, why we call it `query` if it's the `body` ?

File: client/state/login/actions.js [hunk 3]
```diff
@@ -293,8 +294,7 @@ export const loginSocialUser = ( service, token, redirectTo ) => dispatch => {
 			dispatch( {
 				type: SOCIAL_LOGIN_REQUEST_FAILURE,
 				error,
-				service: service,
-				token: token,
+				authInfo: socialInfo
```
Comment (yurynix) [inline client/state/login/actions.js]: Why not keep calling it `socialInfo` ?

File: client/blocks/login/social.jsx (hunk 1 above)
Comment (Tug) [inline client/blocks/login/social.jsx]: Well we could but this format has the advantage of being able to be sent directly to our social login endpoint without any formatting, and it isn't used in any of our components. Let's be lazy for now and try to make it work first ;)

**Comment by yurynix** [comment]
Thank you @Tug 👍  LGTM.

File: client/state/login/actions.js (hunk 3 above)
Comment (Tug) [inline client/state/login/actions.js]: Cause we are already in the `socialAccount` reducer, this property is simply the auth response from the social service that we are storing temporarily, makes more sense to call it `socialAccount.authInfo` in this case

Comment (yurynix) [inline client/state/login/actions.js]: That's creates the illusion, to me, that `authInfo` and `socialInfo` are different, while they are not ;p

**Pull request merged**